### ISSUE TITLE
deCONZ: Waiting for device

### ISSUE THREAD

**Issue by Dinth** [home-assistant/hassio-addons#1240]
Im trying to set up the device path with /dev/serial/by-id/...
```
device: >-
  /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_Gmbh_blahblah
vnc_password: 'blahblah'
```
And it almost never works. I've restarted my deCONZ a couple of dozens of times and only once it picked up the device configured this way.

```
[17:31:42] FATAL: No device /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_blahblahblah-if00 found!
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

But setting up the device this way:

```
device: /dev/ttyACM1
vnc_password: 'blahblah'
```
deCONZ addon works without any problems

Both:
/dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_Gmbh_blahblah and /dev/ttyACM1 are being pasted from Hardware tab in Supervisor settings. My other addons (Zwave, rflink) work fine with /dev/serial/by-id/ paths. Additionally deCONZ running in docker environment outside of HASS also works fine with /dev/serial/by-id/ device path.

**Comment by ThePierce** [comment]
Got the exact same problem in Hassio version 109.1

**Comment by MTrab** [comment]
+1 never got this to work with the /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2141754-if00 path.

Please fix this, as the /dev/ttyACM is randomly assigned on boot - I have multiple USB devices registering as ACM, so I cannot be sure that the number is the same every time :(

**Comment by frenck** [comment]
Please provide system information; like use operating system, hardware, the way things are run and set up.

Thanks! 👍

**Comment by MTrab** [comment]
Hardware: ESXi
OS: Debian Stretch
Setup: Using this guide https://www.home-assistant.io/hassio/installation/#alternative-install-home-assistant-supervised-on-a-generic-linux-host

**Comment by MTrab** [comment]
Also worth noting - after 8 tries to start the deConz addon it succeded using /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2141754-if00

When using /dev/ttyACMx it works every time the addon starts.

**Comment by Dinth** [comment]
> Also worth noting - after 8 tries to start the deConz addon it succeded using /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2141754-if00

That's pretty random. For me it took probably around 10 restarts for deConz to start with /by-id/ device, but then i havent managed to reproduce that with another 20-30 restarts.

**Comment by tocklime** [comment]
I'm also getting this behaviour - have had deconz running happily in portainer on my home assistant machine for a long time. Wanted to move it back under home assistant's control, so followed the migration instructions. It starts up after several (seems a random number, less than 10) tries. (the other times it spends a long time after `INFO: Waiting for device...` then gives up.

I'm running home assistant supervised on a generic host - i've just seen that isn't supported any more. I will have to see about reinstalling everything I suppose.

**Comment by tocklime** [comment]
in those cases where it's not starting right, I see this on the commandline:

```bash
greg@zealot:~$ ls -l /dev/serial/by-id/
total 0
lrwxrwxrwx 1 root root 13 May 28 11:33 usb-1a86_USB2.0-Serial-if00-port0 -> ../../ttyUSB1
lrwxrwxrwx 1 root root 13 May 28 11:33 usb-FTDI_FT230X_Basic_UART_DM01C1TI-if00-port0 -> ../../ttyUSB0
lrwxrwxrwx 1 root root 13 May 28 11:33 usb-Silicon_Labs_CP2102_USB_to_UART_Bridge_Controller_0001-if00-port0 -> ../../ttyUSB2
greg@zealot:~$ docker exec -i -t addon_core_deconz  /bin/bash
root@core-deconz:/# ls -l /dev/serial/by-id/
total 0
lrwxrwxrwx 1 root root 13 May 28 11:33 usb-Silicon_Labs_CP2102_USB_to_UART_Bridge_Controller_0001-if00-port0 -> ../../ttyUSB2
root@core-deconz:/#
```

My host is called zealot. You can see on the host I have 3 links under /dev/serial/by-id, but in the container there is only one.

**Comment by hermannsblum** [comment]
I'm experiencing the same issue, is there a solution in sight?

**Comment by sanderlv** [comment]
Same here! Esxi. If I remove the usv stick and ad it again (physically) it runs 1st time! Else not ... 

I have 3 devices and do not know how to distinguish what is what...

serial:
/dev/ttyS3
/dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2194613-if00
/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_AO3RXEHS-if00-port0
/dev/serial/by-id/usb-0658_0200-if00
/dev/ttyS0
/dev/ttyUSB0
/dev/ttyS1
/dev/ttyACM0
/dev/ttyACM1
/dev/ttyS2

**Comment by Miraculix666** [comment]
Maybe same problem here, Rpi4, deconz on hassio couldn't connect to Raspbee. Issue eventually happend after a round of updates.

deconz log:


Restoring a previous snapshot didn't work, same error with a fresh HassIO installation. When I start a fresh, pure, deconz image from another card, everything is fine.
